web-api/src/personalization/rerank.rs

```diff
@@ -73,35 +74,32 @@ fn rerank_by_tag_weight<'a>(
 /// The `score_weights` determine the ratios of the scores, it is ordered as
 /// `[interest_weight, tag_weight, elasticsearch_weight]`. The final score/ranking per document is
 /// calculated as the weighted sum of the scores.
-pub(super) fn rerank_by_scores(
+pub(super) fn rerank(
     coi_system: &CoiSystem,
     documents: &mut [PersonalizedDocument],
     interests: &[Coi],
     tag_weights: &HashMap<DocumentTag, usize>,
     score_weights: [f32; 3],
     time: DateTime<Utc>,
 ) {
+    let search_scores = documents.iter().map(|doc| (&doc.id, doc.score)).collect();
     let interest_scores = rerank_by_interest(coi_system, documents, interests, time);
     let tag_weight_scores = rerank_by_tag_weight(documents, tag_weights);
 
-    let scores = documents
-        .iter()
-        .map(|document| {
-            let interest_score = interest_scores
-                .get(&document.id)
-                .copied()
-                .unwrap_or_default();
-            let tag_weight_score = tag_weight_scores
-                .get(&document.id)
-                .copied()
-                .unwrap_or_default();
-            score_weights[0] * interest_score
-                + score_weights[1] * tag_weight_score
-                + score_weights[2] * document.score
-        })
-        .collect_vec();
-    for (document, score) in documents.iter_mut().zip(scores) {
-        document.score = score;
+    let scores = rrf(
+        DEFAULT_RRF_K,
+        [
+            (score_weights[0], interest_scores),
+            (score_weights[1], tag_weight_scores),
+            (score_weights[2], search_scores),
```

Comment:
:question:  I'm not sure we can use that weights just like that.

The default weights have been largely determined by the difference in the distributions of the different metricts, which with rrf doesn't matter.

Reply:
We should set everything to one. The weights here weren't thought to be used with rrf. We can remove them completely. Maybe in the future, we could enable another _dev option in case we want to experiment.

Reply:
Be aware that we currently set the weight for tags for personalization to `0` and for semantic search to 1/2 the weight of the other weights.

Which was the main reason I didn't strip it our (or replace it).

Reply:
I now set it to `1, 1, 0` and `1, 1, 0.5` which produces more similar results then what we had before and in turn requires less changes to tests.

Reply:
ok let's go with that